Extending/Packages/Good-practice/index.md

```diff
@@ -0,0 +1,161 @@
+---
+versionFrom: 9.0.0
+meta.Title: "Package development good practice"
+meta.Description: "Information on common defaults and good practice for umbraco package development"
+---
+
+# Good practice and defaults
+_This document provides some guides and notes on package development, and good practice that will help you maintain and support your package through multiple releases and versions of Umbraco. These good practices are not prescriptive, but mearly offer a guide as to what we have found works well, and not-so-well, when developing packages for Umbraco_
+
+## App_Plugins
+Files in the `App_Plugins` folder should be considered immutable, that is they are not something a user of your package should be expecting to change on their site. 
+
+The default delivery method for files to the App_Plugins folder is via a `.targets` file within a package, this means when a website is built the files in this folder are copied over from the nuget cache, and any changes a user might have made will be lost. Equally if the user performs a `dotnet clean` on a solution all files in the App_Plugins folder will likely be deleted. 
+
+If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
+
+## Views
+Views are used to render content on the front end of a website. If your plugin provides a way for the user to present the content publicly you should copy these files to the views folder. 
+
+As the files will still be copied during build you should ensure your target file doesn't overwrite newer or altered files and doesn't delete files on clean. 
+
+_This method isn't perfect and if you do have a package that is heavy on views you might consider building a dotnet template or Razor class library to contain the views so they don't pollute your user's solutions._ 
+
+## License files
+Umbraco products store their licenses in `/umbraco/Licences` and it is considered a good location for third party packages that require license files to also store their licence files there. 
+
+_The default .gitignore for Umbraco templates will include any files in this folder in a repo (while ignoring most of the rest of the Umbraco folder)_
+
+:::note
+The `/umbraco/Licenses` folder does not exist on a fresh installation of Umbraco so you might need to create it before you save your license file to this folder. 
+:::
+
+## Operating system considerations
+Umbraco 9 is a .NET Core application and it can be ran on multiple operating systems (e.g Windows, Linux). When developing packages there are a few things you should be aware of so your package can run on all possible operating systems. 
+
+### Case sensitivity 
```

Comment:
Is using constants to keep the casing the same something we should advise?

Reply:
Yes - added a line